src/commands/list_folder.ts

```diff
@@ -0,0 +1,130 @@
+/* eslint-disable no-console */
+import { arweave } from 'ardrive-core-js';
+import { cliWalletDao, CLI_APP_NAME, CLI_APP_VERSION } from '..';
+import { ArDrive, ArDriveAnonymous } from '../ardrive';
+import {
+	ArFSDAO,
+	ArFSDAOAnonymous,
+	ArFSPrivateFileOrFolderData,
+	ArFSPublicFileOrFolderData,
+	FolderHierarchy
+} from '../arfsdao';
+import { CLICommand } from '../CLICommand';
+import { CommonContext } from '../CLICommand/common_context';
+import { ArDriveCommunityOracle } from '../community/ardrive_community_oracle';
+import { ParentFolderIdParameter, SeedPhraseParameter, WalletFileParameter } from '../parameter_declarations';
+
+new CLICommand({
+	name: 'list-folder',
+	parameters: [ParentFolderIdParameter, SeedPhraseParameter, WalletFileParameter],
+	async action(options) {
+		const context = new CommonContext(options, cliWalletDao);
+		const wallet = await context.getWallet().catch(() => null);
+		const folderId = context.getParameterValue(ParentFolderIdParameter);
+		let folder;
+		let mergedData: (ArFSPrivateFileOrFolderData | ArFSPublicFileOrFolderData)[];
+
+		if (!folderId) {
+			console.log(`Folder id not specified! ${folderId}`);
+			process.exit(1);
+		}
+
+		if (wallet) {
+			const arDrive = new ArDrive(
+				wallet,
+				cliWalletDao,
+				new ArFSDAO(wallet, arweave),
+				new ArDriveCommunityOracle(arweave),
+				CLI_APP_NAME,
+				CLI_APP_VERSION
+			);
+			// Fetch the folder to extract the drive
+			folder = await arDrive.getPrivateFolder(folderId);
+
+			// Fetch all of the folder entities within the drive
+			const driveIdOfFolder = folder.driveId;
+			const allFolderEntitiesOfDrive = await arDrive.getAllFoldersOfPrivateDrive(driveIdOfFolder);
+
+			// Feed entities to FolderHierarchy.setupNodesWithEntity()
+			const hierarchy = FolderHierarchy.newFromEntities(allFolderEntitiesOfDrive);
+			const folderIDs = hierarchy.allFolderIDs();
+
+			// Fetch all file entities within all Folders of the drive
+			const allFileEntitiesOfDrive = await arDrive.getPrivateChildrenFilesFromFolderIDs(folderIDs);
+
+			// Fetch all names of each entity
+			const allEntitiesOfDrive = [...allFolderEntitiesOfDrive, ...allFileEntitiesOfDrive].sort(
+				(a, b) => +a.txId - +b.txId
+			);
+
+			mergedData = allEntitiesOfDrive.map((entity) => {
```

Comment:
Do sorting here. Perhaps on path since it's human readable.

Reply:
e8735f640d062afd36947ba650b39e4b1720d9f1